coordinator/src/main/scala/filodb.coordinator/queryplanner/MultiPartitionPlanner.scala

```diff
@@ -567,20 +567,29 @@ class MultiPartitionPlanner(val partitionLocationProvider: PartitionLocationProv
     }}
   }
 
+  /**
+   *
+   * Extracted to reduce complexity in materializePlanHandleSplitLeaf.
+   *
+   * @param plan the plan to compute the offset for.
+   */
+  private def computeOffsetMs(plan: LogicalPlan): Seq[Long] = plan match {
+    case op: ScalarVectorBinaryOperation => Seq(getOffsetMillis(op).max)
```

Comment:
Why does this need a special handler for `ScalarVerctorBinaryOperation`s? Was that case missed in `getOffsetMillis`?

Reply:
Scalar has an offset of 0 which can be ignored. Eg.
`123 + foo() offset 10m` have offset (0m, 10m), we only care about the offset for foo, so we take the max of the two. This gives better efficiency. 
(and it would fail lots of old test cases for ScalarVectorBinaryOperation if we don't take max)

Reply:
I made it a separate method only because it yells "Cyclomatic complexity of 21 exceeds max of 20." if I don't.

Reply:
One query I think might snag here:
```
foo{} + scalar(foo{} offset 10m)
```

The complexity to address every case might not be worth the benefit-- how do you feel about ditching this extra logic for now? We can always come back and add a `match` like this if needed.

Reply:
Ditching it is much safer. I'll rewrite/remove the test cases.